libs/core/langchain_core/chat_history.py

```diff
@@ -56,6 +57,10 @@ def clear(self):
     messages: List[BaseMessage]
     """A list of Messages stored in-memory."""
 
+    async def amessages(self) -> List[BaseMessage]:
```

Comment:
Yes. That's especially problematic in this part
```python
    def __str__(self) -> str:
        """Return a string representation of the chat history."""
        return get_buffer_string(self.messages)
```
If you call this in async mode, it might block the event loop.